tests/store/model_registry/test_file_store.py

```diff
@@ -1,1364 +1,1341 @@
-        mv2 = self._create_model_version(fs, name=name, source="A/C", run_id=run_id_2)
-        assert mv2.version == 2
-        mv3 = self._create_model_version(fs, name=name, source="A/D", run_id=run_id_2)
-        assert mv3.version == 3
-        mv4 = self._create_model_version(fs, name=name, source="A/D", run_id=run_id_3)
-        assert mv4.version == 4
-
-        def search_versions(filter_string):
-            return [mvd.version for mvd in fs.search_model_versions(filter_string)]
-
-        # search using name should return all 4 versions
-        assert set(search_versions("name='%s'" % name)) == {1, 2, 3, 4}
-
-        # search using run_id_1 should return version 1
-        assert set(search_versions("run_id='%s'" % run_id_1)) == {1}
-
-        # search using run_id_2 should return versions 2 and 3
-        assert set(search_versions("run_id='%s'" % run_id_2)) == {2, 3}
-
-        # search using the IN operator should return all versions
-        assert set(search_versions(f"run_id IN ('{run_id_1}','{run_id_2}')")) == {1, 2, 3}
+            MlflowException, match=f"Could not find registered model with name {name}"
+        ):
+            store.get_registered_model(name)
 
-        # search IN operator is case sensitive
-        assert set(search_versions(f"run_id IN ('{run_id_1.upper()}','{run_id_2}')")) == {2, 3}
 
-        # search IN operator with right-hand side value containing whitespaces
-        assert set(search_versions(f"run_id IN ('{run_id_1}', '{run_id_2}')")) == {1, 2, 3}
+def test_list_registered_model(store, registered_models_names, rm_data):
+    for rm in store.list_registered_models(max_results=10, page_token=None):
+        name = rm.name
+        assert name in registered_models_names, "Not found in registered modules"
+        assert name == rm_data[name]["name"]
 
-        # search using the IN operator with bad lists should return exceptions
-        with pytest.raises(
-            MlflowException,
-            match=(
-                r"While parsing a list in the query, "
-                r"expected string value, punctuation, or whitespace, "
-                r"but got different type in list"
-            ),
-        ) as exception_context:
-            search_versions("run_id IN (1,2,3)")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        assert set(search_versions(f"run_id LIKE '{run_id_2[:30]}%'")) == {2, 3}
+def test_rename_registered_model(store, registered_models_names, rm_data):
+    model_name = registered_models_names[random_int(0, len(registered_models_names) - 1)]
+    # Error cases
+    with pytest.raises(MlflowException, match="Registered model name cannot be empty."):
+        store.rename_registered_model(model_name, None)
+    # test that names of existing registered models are checked before renaming
 
-        assert set(search_versions(f"run_id ILIKE '{run_id_2[:30].upper()}%'")) == {2, 3}
+    for name in registered_models_names:
```

Comment:
Do we need to do this? Why the random selection here? Why not just take the last and first entries from registered_model_names and assert that we can't rename the last as the first?